src/routers/api.ts

```diff
@@ -131,18 +131,42 @@ router.post('/messages', async (req, res, next) => {
     if (req.body.message === undefined) {
       throw new RequestError('Missing message', 400);
     }
-    if (req.body.recipient === undefined) {
+    let senderDestination: string | undefined = undefined;
+    if (typeof req.body.sender === 'string') {
+      if (!req.body.sender.startsWith(peerID)) {
```

Comment:
Do we need to check that the character after `peerID` finishes is either the end of the string, or `/`?

Reply:
Great point, we are currently taking the secondary ID as what follows the peer ID + 1 character. Exactly as you pointed out, that character may or may not match the designated separator.

Reply:
Discussed this with @gabriel-indik offline, and we're going to merge this PR and he will add a follow-on to refine this check to a full match.